openshift-knative-operator/pkg/serving/extension.go

```diff
@@ -7,30 +7,38 @@ import (
 
 	mf "github.com/manifestival/manifestival"
 	"github.com/openshift-knative/serverless-operator/openshift-knative-operator/pkg/common"
+	"github.com/openshift-knative/serverless-operator/openshift-knative-operator/pkg/monitoring"
+	"github.com/openshift-knative/serverless-operator/pkg/client/clientset/versioned"
+	ocpclient "github.com/openshift-knative/serverless-operator/pkg/client/injection/client"
+	"go.uber.org/zap"
 	"k8s.io/apimachinery/pkg/api/resource"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
 	"knative.dev/operator/pkg/apis/operator/v1alpha1"
-
-	"github.com/openshift-knative/serverless-operator/pkg/client/clientset/versioned"
-	ocpclient "github.com/openshift-knative/serverless-operator/pkg/client/injection/client"
 	operator "knative.dev/operator/pkg/reconciler/common"
+	kubeclient "knative.dev/pkg/client/injection/kube/client"
+	"knative.dev/pkg/logging"
 )
 
 const loggingURLTemplate = "https://%s/app/kibana#/discover?_a=(index:.all,query:'kubernetes.labels.serving_knative_dev%%5C%%2FrevisionUID:${REVISION_UID}')"
 
 // NewExtension creates a new extension for a Knative Serving controller.
 func NewExtension(ctx context.Context) operator.Extension {
 	return &extension{
-		ocpclient: ocpclient.Get(ctx),
+		ocpclient:  ocpclient.Get(ctx),
+		kubeclient: kubeclient.Get(ctx),
+		logger:     logging.FromContext(ctx),
 	}
 }
 
 type extension struct {
-	ocpclient versioned.Interface
+	ocpclient  versioned.Interface
+	kubeclient kubernetes.Interface
+	logger     *zap.SugaredLogger
```

Comment:
This is not required. The context passed to `Reconcile` already has a context that can be obtained using `logging.FromContext`.

Reply:
Good to know there is a context otherwise it seems too verbose.